.github/workflows/build.yml

```
@@ -32,9 +35,14 @@ jobs:
           distribution: 'zulu'
           java-version: 19
 
-      - name: Test
+      - name: Full build
+        if: matrix.full-build
         run: ./gradlew build
 
+      - name: File system check
+        if: "!matrix.full-build"
```

I tried a few different things here including:

```yaml
        if: $${{ !matrix.full-build }}
        if: $${{ matrix.full-build != true }}
```

Escaping with quotation marks works but I am not sure why:

https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsif
https://docs.github.com/en/actions/learn-github-actions/expressions

---

Would it be easier to just make checks against `os` instead to avoid introducing a new environment variable? E.g.:

```suggestion
        if: matrix.os != 'ubuntu-latest'
```

And similarly for the `Full build` job.